### ISSUE TITLE
MPI-4.0: MPI_Group_from_session_pset() seems to be broken

### ISSUE THREAD

**Issue by dalcinl** [open-mpi/ompi#10577]
Reproducer:

```python
from mpi4py import MPI

session = MPI.Session.Init()
num_psets = session.Get_num_psets()
for n in range(num_psets):
    name = session.Get_nth_pset(n)
    print(name)
    group = MPI.Group.Create_from_session_pset(session, name)
    group.Free()
session.Finalize()
```

Output:

```
$ mpiexec -n 1 python /tmp/tmp.py 
mpi://WORLD
Traceback (most recent call last):
  File "/tmp/tmp.py", line 8, in <module>
    group = MPI.Group.Create_from_session_pset(session, name)
  File "mpi4py/MPI/Group.pyx", line 210, in mpi4py.MPI.Group.Create_from_session_pset
mpi4py.MPI.Exception: MPI_ERR_ARG: invalid argument of some other kind
```

**Comment by dalcinl** [comment]
@hppritcha You may want to look at this one.

**Comment by hppritcha** [comment]
okay clearly need to add some sessions sanity checks to the ompi-tests repo.

**Comment by awlauria** [comment]
Merged to main and v5.0.x . Closing.
